src/backend/backend/tasks.py

```diff
@@ -0,0 +1,124 @@
+import asyncio
+import io
+import json
+import uuid
+from pathlib import Path
+
+import httpx
+from a2a.types import JSONRPCErrorResponse, SendMessageSuccessResponse
+
+from backend.celery_app import celery_app
+from backend.client import AgentClient
+from backend.db import session_manager
+from backend.logging import logger
+from backend.repositories.agents import AgentRepository
+from backend.schemas import AgentStatus
+from backend.settings import settings
+from backend.storage import storage_client
+from backend.templates import AGENT_CODE_TEMPLATE, DOCKERFILE_TEMPLATE
+
+BASE_URL = settings.A2A_AGENT_URL
+
+
+def _get_agent_repository():
+    with session_manager.session() as session:
+        yield AgentRepository(session)
+
+
+def _update_agent_status(agent_id: uuid.UUID, status: str):
+    logger.info(f"Updating agent {agent_id} status to {status}")
+    for repo in _get_agent_repository():
+        repo.update(agent_id, status=status)
+
+
```

Comment:
❗ I think we need a function to update the `download_available` when it's ready.

```suggestion

def _update_agent_download(agent_id: uuid.UUID, available: bool):
    logger.info(f"Updating agent {agent_id} download_available to {available}")
    for repo in _get_agent_repository():
        repo.update(agent_id, download_available=available)
        
        
```